src/croniter/croniter.py

```diff
@@ -790,11 +823,12 @@ def _expand(cls, expr_format, hash_id=None, second_at_beginning=False, from_time
                         pass
 
                     if t in cls.LOWMAP[i] and not (
-                        # do not support 0 as a month either for classical 5 fields cron
-                        # or 6fields second repeat form
+                        # do not support 0 as a month either for classical 5 fields cron,
+                        # 6fields second repeat form or 7 fields year form
                         # but still let conversion happen if day field is shifted
                         (i in [2, 3] and len(expressions) == 5) or
-                        (i in [3, 4] and len(expressions) == 6)
+                        (i in [3, 4] and len(expressions) == 6) or
+                        (i in [2, 3, 4] and len(expressions) == 7)
```

Comment:
I have no idea why should check second or forth field depending on length of expressions.
Please correct this.